bindings/python/tests/test_parse.py

```diff
@@ -0,0 +1,332 @@
+    file_contents = ""
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    assert type(parsed_info) is dict
+    assert type(parsed_info["members"]) is list
+    assert len(parsed_info["members"]) == 0
+
+
+def test_translation_unit(tmp_path):
+    file_contents = ""
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    assert parsed_info["kind"] == "TRANSLATION_UNIT"
+    assert parsed_info["depth"] == 0
+    assert parsed_info["line"] == 0
+    assert parsed_info["column"] == 0
+    assert parsed_info["name"] == str(tmp_path / "file.hpp")
+
+
+def test_namespace(tmp_path):
+    file_contents = "namespace a_namespace {}"
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    namespace = parsed_info["members"][0]
+
+    assert namespace["kind"] == "NAMESPACE"
+    assert namespace["name"] == "a_namespace"
+
+
+def test_namespace_ref(tmp_path):
+    file_contents = """
+    #include <ostream>
+    std::ostream anOstream;
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    var_decl = parsed_info["members"][0]
+    namespace_ref = var_decl["members"][0]
+
+    assert namespace_ref["kind"] == "NAMESPACE_REF"
+    assert namespace_ref["name"] == "std"
+
+
+def test_struct_decl(tmp_path):
+    file_contents = "struct AStruct {};"
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+
+    assert struct_decl["kind"] == "STRUCT_DECL"
+    assert struct_decl["name"] == "AStruct"
+
+
+def test_cxx_base_specifier(tmp_path):
+    file_contents = """
+    struct BaseStruct {};
+    struct DerivedStruct: public BaseStruct {};
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    child_struct_decl = parsed_info["members"][1]
+    cxx_base_specifier = child_struct_decl["members"][0]
+
+    assert cxx_base_specifier["kind"] == "CXX_BASE_SPECIFIER"
+    assert cxx_base_specifier["access_specifier"] == "PUBLIC"
+    assert cxx_base_specifier["name"] == "struct BaseStruct"
+
+
+def test_cxx_method(tmp_path):
+    file_contents = """
+    struct AStruct {
+        void aMethod() {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    cxx_method = struct_decl["members"][0]
+
+    assert cxx_method["kind"] == "CXX_METHOD"
+    assert cxx_method["result_type"] == "void"
+    assert cxx_method["name"] == "aMethod"
+
+
+def test_var_decl(tmp_path):
+    file_contents = "int anInt = 1;"
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    var_decl = parsed_info["members"][0]
+
+    assert var_decl["kind"] == "VAR_DECL"
+    assert var_decl["element_type"] == "Int"
+    assert var_decl["name"] == "anInt"
+
+
+def test_type_ref(tmp_path):
```

Comment:
This test could be simpler, or else the name is wrong

Reply:
Something like this?
```cpp
    typedef size_type;
    size_type size;
```

Reply:
o.O Is that valid C++?

Reply:
Sorry, this:
```cpp
    typedef unsigned size_type;
    size_type size;
```

Reply:
Is that a type ref? Or is a type-ref a type reference (as in `int&` and not just `int`)?

Reply:
From cindex:
```py
# A reference to a type declaration.
#
# A type reference occurs anywhere where a type is named but not
# declared. For example, given:
#   typedef unsigned size_type;
#   size_type size;
#
# The typedef is a declaration of size_type (CXCursor_TypedefDecl),
# while the type of the variable "size" is referenced. The cursor
# referenced by the type of size is the typedef for size_type.
CursorKind.TYPE_REF = CursorKind(43)
```